test/python/circuit/test_equivalence.py

```diff
@@ -207,8 +253,8 @@ def test_add_equivalence(self):
         entry = eq_lib.get_entry(gate)
 
         self.assertEqual(len(entry), 2)
-        self.assertEqual(entry[0], second_equiv)
-        self.assertEqual(entry[1], first_equiv)
+        self.assertEqual(entry[0], first_equiv)
+        self.assertEqual(entry[1], second_equiv)
```

Comment:
This test probably ought to be safe in the case of re-ordering of the outputs, since we're not guaranteeing any particular output order.  Some sort of sort into canonical order would be good.

Reply:
Will do.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/8790/commits/299515657e1355b20376124ebc72761c81748fe7.